src/d.ts

```diff
@@ -36,6 +36,28 @@ export function isHNode(child: DNode): child is HNode {
 	return Boolean(child && (typeof child !== 'string') && child.type === HNODE);
 }
 
+/**
+ * Generic decorate function for DNodes, the modified in place based on the provided predicate
+ * and modifier functions.
+ *
+ * If no predicate is supplied then the modifier will be executed on all nodes.
+ */
+export function decorate(dNodes: DNode | DNode[], modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => void): DNode | DNode[] {
+	let nodes = Array.isArray(dNodes) ? [ ...dNodes ] : [ dNodes ];
+	while (nodes.length) {
+		const node = nodes.pop();
+		if (node) {
+			if (!predicate || predicate(node)) {
```

Comment:
"The nodes are modified in place" is a bit clearer that "the modified in place". I guess I just didn't know what the intent of the code is (decorators in the context of higher order functions I assume?)

So basically the intent to to traverse a tree of nodes, flatten them, and apply a modifier and/or predicate to each node?